mini-processor/src/main/java/mini/processor/ReducersGenerator.kt

```diff
@@ -1,84 +1,215 @@
+            val continuationTypeParameter = (function.parameters.last().asType() as DeclaredType)
+                .typeArguments[0].asTypeName() as WildcardTypeName
+            returnTypeName = continuationTypeParameter.inTypes.first()
+            parameters = function.parameters.dropLast(1).map { it.asType().asTypeName() }
+        } else {
+            returnTypeName = function.returnType.asTypeName()
+            parameters = function.parameters.map { it.asType().asTypeName() }
+        }
+
+        if (returnTypeName == UNIT) {
+            isPure = false
+            actionTypeName = function.parameters[0].asType().asTypeName().safeAnyTypeName()
+            compilePrecondition(
+                check = parameters.size == 1,
+                message = "Expected exactly one action parameter",
+                element = element
+            )
+        } else {
+            isPure = true
+            compilePrecondition(
+                check = parameters.size == 2,
+                message = "Expected exactly two parameters, ${container.stateTypeName} and action",
+                element = element
+            )
+            val stateTypeName = parameters[0]
+            actionTypeName = parameters[1].safeAnyTypeName()
+
+            compilePrecondition(
+                check = stateTypeName == container.stateTypeName,
+                message = "Expected ${container.stateTypeName} as first state parameter",
+                element = element
+            )
+
+            compilePrecondition(
+                check = returnTypeName == container.stateTypeName,
+                message = "Expected ${container.stateTypeName} as return value",
+                element = element
+            )
+        }
+
+    }
+
+    fun generateCallBlock(containerParam: String, actionParam: String): CodeBlock {
+
+        val receiver = if (container.isStatic) {
+            CodeBlock.of("%T.${function.simpleName}", container.typeName)
+        } else {
+            CodeBlock.of("${containerParam}.${function.simpleName}")
+        }
+
+        val call = if (isPure) {
+            CodeBlock.of("(${containerParam}.state, $actionParam)")
+        } else {
+            CodeBlock.of("($actionParam)")
+        }
+
+        return if (isPure) {
+            CodeBlock.builder()
+                .add("${containerParam}.setState(")
+                .add(receiver)
+                .add(call)
+                .add(")")
+                .build()
+        } else {
+            CodeBlock.builder()
+                .add(receiver)
+                .add(call)
+                .build()
+        }
+    }
+}
+
+class ContainerModel(element: Element) {
+    val typeName: TypeName
+    val stateTypeName: TypeName
+    val isStatic: Boolean
+
+    init {
+        compilePrecondition(
+            check = element.kind == ElementKind.CLASS,
+            message = "Reducers must be declared inside StoreContainer classes",
```

Comment:
I think that this should be `StateContainer`
```suggestion
            message = "Reducers must be declared inside StateContainer classes",
```

Reply:
Solved in b6f07929daa04d17efee1757805039f0a85e84b6